js/views/userPage/UserPage.js

```diff
@@ -18,12 +20,16 @@ export default class extends baseVw {
     super(options);
     this.options = options;
 
+    this.ownPage = this.model.id === app.profile.id;
+
     this.state = options.state || 'store';
+    if (this.state === 'store' && !this.ownPage && !this.model.get('vendor')) {
```

Comment:
Can we please handle this in the router by showing the Page Not Found page? It can be done by adding this to line 196:

```
      if (state === 'store' && !profile.get('vendor') &&
        profile.id !== app.profile.id) {
        this.pageNotFound();
        return;
      }
```

That's more consistent with how we're handling when you put in a route to a non-existent / active page. I've confirmed with @morebrownies that is the way we want to go.